### PR TITLE
Bump to v0.36.0 - implement describe function

### PR DESCRIPTION
**Pull request by lkarthee**
Continuation of bump polars v0.36.2 work.

I have implemented describe function - it's split between elixir and rust. Tests still fail due to null type. It works if all the data  is numeric. 

Please review and i will fix things for non numeric later. 

```elixir
df = Explorer.DataFrame.new(a: [5,6,7], b: [1, 2, 3])
df_metrics = DF.describe(df)
```

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -5630,7 +5630,47 @@ defmodule Explorer.DataFrame do
   def describe(df, opts \\ []) do
     opts = Keyword.validate!(opts, percentiles: nil)
 
-    Shared.apply_impl(df, :describe, [opts[:percentiles]])
+    if Enum.empty?(df.names) do
+      raise ArgumentError, message: "cannot describe a DataFrame without any columns"
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:5634]:
```suggestion
      raise ArgumentError, "cannot describe a DataFrame without any columns"
```

File: lib/explorer/data_frame.ex [hunk 2]
```diff
@@ -5630,7 +5630,47 @@ defmodule Explorer.DataFrame do
...
+    metrics = ["count", "null_count", "mean", "std", "min"]
+    columns = for m <- metrics, c <- df.names, do: "#{m}:#{c}"
+    p_metrics = for p <- percentiles, do: "#{trunc(p * 100)}%"
+    metrics = metrics ++ p_metrics ++ ["max"]
+
+    p_columns = for p <- p_metrics, c <- df.names, do: "#{p}:#{c}"
+    max_columns = for c <- df.names, do: "max:#{c}"
+    columns = columns ++ p_columns ++ max_columns
+
+    df_metrics = Shared.apply_impl(df, :describe, [percentiles])
```
[the remark this answers is not in the document]
Comment (lkarthee) [reply lib/explorer/data_frame.ex:5647]: I had this rust func which was almost complete. I felt any future changes from python implementation from polars can be easily translated  to rust. 

Should I fix elixir func instead for numeric, temporal and duration types ?

[the remark this answers is not in the document]
Comment (josevalim) [reply lib/explorer/data_frame.ex:5647]: To me it is the opposite, I would prefer to move this pure Elixir as much as possible. The fewer callbacks, the better. :)

[the remark this answers is not in the document]
Comment (lkarthee) [reply lib/explorer/data_frame.ex:5647]: elixir 😎 function it is then..

**Comment by lkarthee** [comment]
Finally green ..

**Review by philss — APPROVED** [review]
LGTM!

I think we could simplify with the usage of structs (Polars dtype introduced recently to Explorer). But I don't know yet how to do, so we can improve later.

**Review by josevalim — APPROVED** [review]
Beautiful. I didn't merge it because it is @philss branch, so I will let him do it. :D

File: lib/explorer/data_frame.ex [hunk 3]
```diff
@@ -5608,29 +5608,80 @@ defmodule Explorer.DataFrame do
...
+        std_exprs = for c <- stat_cols, do: {"std:#{c}", Series.standard_deviation(x[c])}
+        min_exprs = for c <- stat_cols, do: {"min:#{c}", Series.min(x[c])}
+        max_exprs = for c <- stat_cols, do: {"max:#{c}", Series.max(x[c])}
+
+        counts_exprs ++
+          nil_counts_exprs ++
+          mean_exprs ++ std_exprs ++ min_exprs ++ percentile_exprs ++ max_exprs
+      end)
+
+    metric_row = df_metrics |> to_rows() |> hd()
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:5660]:
```suggestion
    metric_row = df_metrics |> collect() |> to_rows() |> hd()
```

**Pull request merged**